src/main/cpp/chassis/headingStates/MaintainHeading.h

```diff
@@ -28,4 +29,9 @@ class MaintainHeading : public ISwerveDriveOrientation
     ~MaintainHeading() = default;
 
     void UpdateChassisSpeeds(ChassisMovement &chassisMovement) override;
+
+private:
+    bool m_prevTranslatinOrStrafing = false;
+    frc::PIDController m_fineController{5.0, 0.5, 0.0};
+    frc::PIDController m_coarseController{5.0, 0.5, 0.0};
```

Comment:
Should probably clean this code up to only use the one controller, but I am fine leaving it here for now. @witcpalek

Reply:
I'll do it tonight